### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ]  In this project, I finished creating the remaining classes for my school library and create the associations between them.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-    super(name, age, parent_permission: parent_permission)
-    @classroom = classroom
-  end
-
-  def play_hooky
-    '¯\(ツ)/¯'
+  def initialize(name)
+    @name = name
   end
 end
```
Comment (DammyShittu) [inline student.rb]: - [ ] Do we really need to delete these lines of code? Kindly re-add them as they were. We need them so we can be able to create a Student who inherits from the Person class.

- [ ] Kindly add the other end of the association with Classroom class.

  ```
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: person.rb [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
   def initialize(name)
-    super
     @name = name
-  end
-
-  def correct_name
-    @name
+    @rentals = []
   end
 end
```
Comment (DammyShittu) [inline person.rb]: - [ ] For these associations to work we need to be able to create a Person in the first place. Kindly go back to your previous code ([visit this file in this PR](https://github.com/Ibtisam34/OOP-school-library/pull/1/files#diff-47720676886debe217f2360ee91e7dcbfca82989297cf253a11792456731033f) and get your code back) and re-add the missing lines of code that allow us to create a Person.

  Also, bring back all the missing methods this class once had in earlier reviews. We are building on previous code, there is no need to delete this amount of code.

- [ ] Be sure to maintain the `@rentals` instance variable in the `initialize` method.

  ```
  def initialize(...)
    ...
    @rentals = [ ]
  end
  ```

- [ ] Kindly add the method for establishing an association with the Book and Rental classes.

  ```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
  ```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accesor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
```
Comment (DammyShittu) [inline book.rb:4]: - [ ] Kindly initialize `@rentals` and assign an empty array to it.

  ```
  rentals = [ ]
  ```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  end
+end
```
Comment (DammyShittu) [inline book.rb]: - [ ] Kindly add the method for associating with the Person and Rental classes.

  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Ibtisam34,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 3 above)
Comment (Whoistolu) [reply book.rb:4]: - Well resolved @Ibtisam34

File: book.rb (hunk 4 above)
Comment (Whoistolu) [reply book.rb]: - Well resolved @Ibtisam34

File: person.rb (hunk 2 above)
Comment (Whoistolu) [reply person.rb]: - Well resolved @Ibtisam34

**Review by Whoistolu — APPROVED** [review]
### Hi @Ibtisam34 

Great implementing the recommendations by the previous reviewers 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**